src/audio_core/sink/null_sink.h

```diff
@@ -15,14 +31,15 @@ class NullSink final : public Sink {
     explicit NullSink(std::string_view) {}
     ~NullSink() override = default;
 
-    SinkStream* AcquireSinkStream([[maybe_unused]] Core::System& system,
-                                  [[maybe_unused]] u32 system_channels,
-                                  [[maybe_unused]] const std::string& name,
-                                  [[maybe_unused]] StreamType type) override {
-        return &null_sink_stream;
+    SinkStream* AcquireSinkStream(Core::System& system, u32, const std::string&,
+                                  StreamType type) override {
+        if (null_sink.get() == nullptr) {
```

Comment:
Equality with nullptr can be done directly with smart pointers

```suggestion
        if (null_sink == nullptr) {
```